cmake/kokkos_tpls.cmake

```diff
@@ -44,7 +44,8 @@ IF(KOKKOS_ENABLE_HIP AND NOT KOKKOS_CXX_COMPILER_ID STREQUAL HIPCC)
 ELSE()
   SET(ROCM_DEFAULT OFF)
 ENDIF()
-KOKKOS_TPL_OPTION(ROCM    ${ROCM_DEFAULT})
+KOKKOS_TPL_OPTION(ROCM      ${ROCM_DEFAULT})
+KOKKOS_TPL_OPTION(ROCTHRUST ${ROCM_DEFAULT})
```

Comment:
We need to think carefully about this.  My initial reaction is that we should have the enable rocthrust option ON by default when HIP is enabled and have the user explicitly disable it if he still wants to build without it.
For reference
https://github.com/arborx/ArborX/blob/3cb2fa329e4ca7beb019612ada8e571d80af0f74/CMakeLists.txt#L18-L30

Reply:
@dalg24 Something like this?